learning/tour-of-beam/frontend/lib/cache/unit_progress.dart

```diff
@@ -66,38 +114,13 @@ class UnitProgressCache extends Cache {
     return UnitCompletion.uncompleted;
   }
 
-  bool isUnitCompleted(String? unitId) {
-    return getCompletedUnits().contains(unitId);
-  }
-
-  Future<void> updateCompletedUnits() async {
-    final sdkId = GetIt.instance.get<AppNotifier>().sdkId;
-    if (sdkId != null) {
-      await _loadCompletedUnits(sdkId);
-    }
-  }
-
-  Set<String> getCompletedUnits() {
-    if (_future == null) {
-      unawaited(updateCompletedUnits());
-    }
-
-    return _completedUnitIds;
-  }
-
-  Future<void> _loadCompletedUnits(String sdkId) async {
-    _future = client.getUserProgress(sdkId);
-    final result = await _future;
+  // Snippets
 
-    _completedUnitIds.clear();
-    if (result != null) {
-      for (final unitProgress in result.units) {
-        if (unitProgress.isCompleted) {
-          _completedUnitIds.add(unitProgress.id);
-        }
-      }
+  Map<String, String?> getUnitSnippets() {
```

Comment:
All uses of this method only pick one snippet. There should be a simpler way to do that without iterating them all and filling the map.

Have a map of `id -> UnitProgress`, fill it in `_loadUnitProgress()`, then allow getting `UnitProgress` objects instead of snippet IDs?